src/Microsoft.PowerShell.Commands.Utility/commands/utility/GetHash.cs

```diff
@@ -0,0 +1,349 @@
+                        }
+                        else
+                        {
+                            ItemNotFoundException pathNotFound =
+                                new ItemNotFoundException(
+                                    newPath,
+                                    "PathNotFound",
+                                    SessionStateStrings.PathNotFound);
+                            ErrorRecord errorRecord = new ErrorRecord(pathNotFound,
+                                "FileNotFound",
+                                ErrorCategory.ObjectNotFound,
+                                path);
+                            WriteError(errorRecord);
+                        }
+                    }
+                    break;
+            }
+
+            foreach (string path in pathsToProcess)
+            {
+                byte[] bytehash = null;
+                String hash = null;
+
+                try
+                {
+                    Stream openfilestream = File.OpenRead(path);
+                    bytehash = hasher.ComputeHash(openfilestream);
+
+                    hash = BitConverter.ToString(bytehash).Replace("-","");
+                    WriteHashResult(Algorithm, hash, path);
+                }
+                catch
+                {
+
+                }
+            }
+        }
+
+        /// <summary>
+        /// Create FileHashInfo object and output it
+        /// </summary>
+        private void WriteHashResult(String Algorithm, String hash, String path)
+        {
+            FileHashInfo result = new FileHashInfo();
+            result.Algorithm = Algorithm;
+            result.Hash = hash;
+            result.Path = path;
+            WriteObject(result);
+        }
+
+        /// <summary>
+        /// Parameter set names
+        /// </summary>
+        private const string PathParameterSet = "Path";
+        private const string LiteralPathParameterSet = "LiteralPath";
+        private const string StreamParameterSet = "StreamParameterSet";
+
+    }
+
+    /// <summary>
+    /// Base Cmdlet for cmdlets which deal with crypto hashes
+    /// </summary>
+    public class HashCmdletBase : PSCmdlet
+    {
+        /// <summary>
+        /// Algorithm parameter
+        /// The hash algorithm name: "SHA1", "SHA256", "SHA384", "SHA512", "MD5"
+        /// </summary>
+        /// <value></value>
+        [Parameter(Position = 1)]
+        [ValidateSet(HashAlgorithmNames.SHA1,
+                     HashAlgorithmNames.SHA256,
+                     HashAlgorithmNames.SHA384,
+                     HashAlgorithmNames.SHA512,
+                     HashAlgorithmNames.MD5)]
+        public String Algorithm
+        {
+            get
+            {
+                return _Algorithm;
+            }
+            set
+            {
+                // A hash algorithm name is case sensitive
+                // and always must be in upper case
+                _Algorithm = value.ToUpper();
```

Comment:
Never mind - `ValidateSet` ignores case.

Maybe not applicable here, but it's often useful to store the exact argument a user specifies so error messages or whatever look like what they wrote. Of course that means you couldn't use `switch` as easily, but I don't think that's a big deal.